Cookbook/Technical-Documents/SwiftGen.md

```diff
@@ -0,0 +1,32 @@
+## SwiftGen
+
+SwiftGen allows plenty of possibilities to automatically generate code based on resources we store in our project. Following sections will describe which types of resources are currently integrated with SwiftGen and how to use them.
+
+### Image assets
+
+Using icons and images with our SwiftGen integration is really easy. The only thing you have to do is to add image to `BabylonDesignLibrary/DesignLibrary/Assets.xcassets` and then you can just use it by calling `appearance.tokens.icons.image(.myNewIcon)`. Before compilation of `BabylonDesignLibrary` SwiftGen is executed and it regenerates identifiers.
+If you name your icon with camel case style the same name will be generated. If you choose sneak case style e.g. `my_new_icon` SwiftGen will process it and will produce camel case identifier `myNewIcon`.
+
+#### Overriding icons in white label apps
+
+This is the method generated by SwiftGen which is used to access UIImages:
+```
+public static func image(_ name: Name) -> UIImage {
+	return image(name, bundle: .main) 
+}
+
+public static func image(_ name: Name, bundle: Bundle) -> UIImage {
+	let iconsBundle = Bundle(for: IconsBundleToken.self)
+	guard let result = UIImage(named: name.rawValue, in: bundle, compatibleWith: nil)
+		?? UIImage(named: name.rawValue, in: iconsBundle, compatibleWith: nil) else {
+		 fatalError("Unable to load image named \(name).") 
+	}
+	return result
+}
+```
+The method without specifying bundle automatically chooses `Bundle.main`. The algorithm is following:
+1. Method checks if the icon is added in assets catalog in the main bundle. In most cases it will be white label app bundle (this allows overriding the icon for specific app).
+2. If the icon is not there it takes icon from DesignLibrary assets catalog.
+3. In other case `fatalError` is called, but this situation is impossible because icon has to be in DesignLibrary assets catalog because SwiftGen created identifier for it. (If icon is deleted SwiftGen will regenerate file without identifier for that icon).
+
+When some app overrides icon with the same name in its assets catalog but it still would like to use icon from DesignLibrary in other case it can specify bundle: `Bundle(for: IconsBundleToken.self)` and it will enforce that image from DesignLibrary assets catalog will retrieved.
```

Comment:
```suggestion
When an app overrides an icon with the same name in its assets catalog but you would still like to use the icon from DesignLibrary in some cases, you can explicitly pass `DesignLibrary.bundle` to enforce that the image is loaded from the DesignLibrary assets catalog.
```

Note: Would actually have `DesignLibrary.bundle` (see `DesignLibrary.swift` line 35) which feels easier to understand and read than `Bundle(for: IconsBundleToken.self)`.

It would actually be nice to update the main repo to get rid of `IconsBundleToken` in our custom SwiftGen template, and use `DesignLibrary.bundle` instead.